app/controllers/api/v0/service_plans_controller.rb

```diff
@@ -6,14 +6,25 @@ class ServicePlansController < ApplicationController
 
       def order
         service_plan = model.find(service_plan_id)
-        task_id = service_plan.order(order_params)
-        render :json => {:task_id => task_id}
+        task = Task.create!(:tenant => service_plan.tenant, :status => "started")
+
+        #TODO: Publish a message with service plan ordering messaging client and simply
+        # return the task id, let the messaging client handle updating the task.
+        order_service_plan_and_update_task(task, service_plan)
+
+        render :json => {:task_id => task.id}
       rescue ActiveRecord::RecordNotFound
         head :bad_request
       end
 
       private
 
+      def order_service_plan_and_update_task(task, service_plan)
+        task.context = service_plan.order(order_params)
+        task.status = "completed"
```

Comment:
@eclarizio In my testing I have seen that the service_plan.order sends a request to OCP and comes back right away without waiting for the provision to end, so is it appropriate to mark the status as completed here. Unless its just a place holder.

Reply:
@mkanoor this is out of scope of this change